doc/rst/tools/mason/mason.rst

```diff
@@ -95,21 +79,69 @@ When invoked, ``mason build [ options ]`` will do the following:
     - Run the executable built above out of ``target/``, if it exists.
     - All options not recognized by ``mason`` will be forwarded to the executable.
 
-For example, after ``mason build && mason run [ options ]``, the project directory appears as so::
+For example, after ``mason run --build [ options ]``, the project directory appears as follows::
+
+  MyProject
+   │
+   ├── Mason.lock
+   ├── Mason.toml
+   ├── example
+   ├── src
+   │   └── myProject.chpl
+   ├── target
+   │   ├── debug
+   │   │   └── myProject
+   │   ├── example
+   │   └── test
+   └── test
+
+
+As you can see, new files have been added to the project, the first of which
+is the ``Mason.lock``. You can think of this file as a snapshot of a single
+run of the program. This file "locks" in the settings in which the program
+ran upon invocation of ``mason run``. This file can be generated manually
+with the ``mason update`` command. ``mason update`` will read the ``Mason.toml``,
+resolve dependencies, and generate the ``Mason.lock`` based on it's contents.
+
+The ``target/`` directory is where Mason stores all the binaries related to your project.
+These could be binaries for the main source code as well as examples and tests. There are two types of
+targets for building. The example above places the project binary into
```

Comment:
wordsmith:

> The example above places the project binary into
``target/debug/`` because that is the default location of project binary given
no other arguments

> The default location of a project binary is \`\`target/debug/\`\`, as shown in the example above.